docs/website/root/mithril/threat-model.md

```diff
@@ -0,0 +1,293 @@
+
+:::info To do
+
+Missing: the currently recommended relay (reverse proxy)
+
+:::
+
+#### Deployment architecture
+
+This document is specifically targeting the standard deployment architecture where a mithril signer runs next to the block producing node, while access to the mithril aggregator is only done through a relay.
+
+[![Mithril - Container View](./images/deployment-architecture.jpg)](./images/deployment-architecture.jpg)
+
+### External Dependencies
+
+Listing dependencies for mithril-signer
+```
+% cargo-deps-list -e normal -p mithril-signer 
+...
+slog-json v2.6.1 {default}
+slog-scope v4.4.0
+arc-swap v1.6.0
+
+Total dependencies: 267
+```
+
+
+
+<!-- * [libp2p](https://docs.libp2p.io/)
+    * https://docs.libp2p.io/concepts/security/security-considerations/
+    * [DoS Mitigation](https://docs.libp2p.io/concepts/security/dos-mitigation/) -->
+    
+* SPOs infrastructure:
+    * Block producing host configuration
+    * Relay hosts configuration
+    * Firewall/private network
+
+### Entry Points 
+
+* mithril-relay P2P TCP ports
+* Aggregator HTTP port
+* mithril-relay HTTP port
+
+## Assumptions 
+
+The core Mithril protocol is considered safe and its analysis is out of scope for this document, it is computationally infeasible to:
+* Forge a valid aggregate signature from forged signing keys
+* Forge individual signatures impersonating one of the signers
+
+## Assets
+
+For each asset we first identify what protection is required: Availability, Confidentiality, Integrity ie. the [CIA Triad](https://www.splunk.com/en_us/blog/learn/cia-triad-confidentiality-integrity-availability.html)
+
+:::info Note
+
+I follow Pascal's suggestion to identify assets as processes rather than particular resources, except in the case of actual pieces of data (eg. keys). The latter should be listed in the threats section and their protection mitigated
+
+:::
+
+We then identify threats and countermeasures
+
+#### KES private keys
+
+* KES key is present only on BP Node but needs to be shared with both the cardano-node process and the mithril-signer process
+* KES keys are needed by mithril-signer in order to sign verification key along with an operational certificate which authenticates the key from this stake pool id
+* This signing happens at every epoch
+    * **availability**: Yes (If KES key is unavailable then signing cannot proceeed)
+    * **confidentiality**: Yes (Capturing KES private keys allow an attacker to impersonate a registered SPO on-chain and produce blocks on his behalf)
+    * **integrity**: Yes (Rotating a compromised KES key is a time-consuming process, not sure if it's even possible if the KES period has not passed completely? See https://github.com/input-output-hk/cardano-node-wiki/blob/main/docs/stake-pool-operations/7_KES_period.md)
```

Comment:
Yes, KES keys can be rotated when needed, even if the KES period has not expired yet.

Reply:
We should write this sentence into the threat model then.

Reply:
I modified the sentence with the fact that KES keys even if they have not expired :+1: